django/conf/__init__.py

```diff
@@ -30,6 +31,35 @@
 )
 
 
+def add_script_prefix(value):
+    """
+    Prefix header value SCRIPT_NAME to ``value`` under certain circumstances.
+
+    Useful for when the app is being served at a subpath and manually prefixing
+    that subpath to STATIC_URL and STATIC_ROOT in settings is inconvenient
+    (i.e., a deployment concern). Refs #25598.
+    """
+    try:
+        URLValidator()(value)
+        return value
+    except (ValidationError, AttributeError):
+        pass
+    try:
+        value = value.strip()
+    except AttributeError:
+        return value
+    if value.startswith('/'):
+        return value
+    from django.urls import get_script_prefix  # circular dependency if top-level import
+    prefix = get_script_prefix()
+    if not prefix or prefix == '/':
+        return value
+    # try to detect if the prefix is already prefixed appropriately.
+    if value.startswith(prefix) and value != prefix:
```

Comment:
Does this check make sense now that we only handle it if it starts with a relative path? (ie if the prefix where there it would start with a slash anyways)

Reply:
I don't have a ton of experience in Nginx/Apache configuration, but from all of the examples I have been able to find, it looks like the `SCRIPT_NAME` header (the prefix) *does* always begin with a forward slash. So yes, if we can assume that that prefix always begins with a forward slash, then we can remove lines 58, 59. Do you feel confident that we can always assume that the prefix will itself have a slash prefix?

Reply:
If we decide that we can make this assumption, then I will append the newest commit from [jrwdunham:ticket_25598-assume-script-name-slash-prefix](https://github.com/jrwdunham/django/tree/ticket_25598-assume-script-name-slash-prefix) to this PR's branch.

Reply:
@jrwdunham I'm not an expert here, but from what I read the true assumption is that `SCRIPT_NAME` **never** ends with trailing slash. So in case you do not have subfolder, `SCRIPT_NAME` should be just empty/unset.

Reply:
@jrwdunham I think you can drop this if, yes